R/RunDiagnostics.R

```diff
@@ -75,54 +71,85 @@
-                                 inclusionStatisticsFolder = NULL,
-                                 exportFolder,
-                                 databaseId,
-                                 databaseName = databaseId,
-                                 databaseDescription = databaseId,
-                                 cdmVersion = 5,
-                                 runInclusionStatistics = TRUE,
-                                 runIncludedSourceConcepts = TRUE,
-                                 runOrphanConcepts = TRUE,
-                                 runTimeDistributions = TRUE,
-                                 runVisitContext = TRUE,
-                                 runBreakdownIndexEvents = TRUE,
-                                 runIncidenceRate = TRUE,
-                                 runCohortOverlap = TRUE,
-                                 runCohortCharacterization = TRUE,
-                                 covariateSettings = createDefaultCovariateSettings(),
-                                 runTemporalCohortCharacterization = TRUE,
-                                 temporalCovariateSettings = createTemporalCovariateSettings(
-                                   useConditionOccurrence = TRUE,
-                                   useDrugEraStart = TRUE,
-                                   useProcedureOccurrence = TRUE,
-                                   useMeasurement = TRUE,
-                                   temporalStartDays = c(-365, -30, 0, 1, 31),
-                                   temporalEndDays = c(-31, -1, 0, 30, 365)
-                                 ),
-                                 minCellCount = 5,
-                                 incremental = FALSE,
-                                 incrementalFolder = file.path(exportFolder, "incremental")) {
-
-  # collect arguments that were passed to cohort diagnostics at initiation
-  argumentsAtDiagnosticsInitiation <- formals(runCohortDiagnostics)
+executeDiagnostics <- function(cohortDefinitionSet,
+                               exportFolder,
+                               databaseId,
+                               connectionDetails = NULL,
+                               connection = NULL,
+                               cdmDatabaseSchema,
+                               tempEmulationSchema = getOption("sqlRenderTempEmulationSchema"),
+                               cohortDatabaseSchema,
+                               cohortTable = "cohort",
+                               cohortTableNames = CohortGenerator::getCohortTableNames(cohortTable = cohortTable),
+                               vocabularyDatabaseSchema = cdmDatabaseSchema,
+                               cohortIds = NULL,
+                               databaseName = databaseId,
+                               databaseDescription = databaseId,
+                               cdmVersion = 5,
+                               runInclusionStatistics = TRUE,
```

Comment:
could you reorder the parameter to be as identical as possible to the original runCohortDiagnostics

e.g. databaseId and databaseDescription are now separated 

i think it would make it easier for the transition

Reply:
The parameters have actually changed quite significantly so this isn't really possible; this is one reason for the change of name - to highlight that there is a big difference. I have moved database name and description to be next to the databaseId but Its best practice to keep required parameters first and ones with default values after it. With functions like this users should also try and use all named parameters because there are so many.

Reply:
I see; i am fine with either way you want to take it.